### DISCUSSION TITLE
[Unknown Mod][Kingdom Overview screen] Additional symbols are displayed according to the different possibilities of construction for creatures' dwellings

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3577] [Ideas]
Additional symbols are displayed according to the different possibilities of construction for creatures' dwellings in the Kingdom Overview screen.

Screens are for explanation only.
(For Picture 001, I used the Kingdom Overview screen from this issue: https://github.com/ihhub/fheroes2/issues/1844.)

**Kingdom Overview screen, Town/Castle screen.**

In fheroes2, a nice feature allow to build the base creatures' dwellings directly in the Towns/Castles screen.
But it is not possible to build the upgraded creatures' dwellings once the base creatures' dwellings have been built.
This remedie this problem and add an enhancement to the construction of the pre-required buildings for creatures' dwellings.

Picture 001.
![Kingdom info build dwelling H2 03a](https://user-images.githubusercontent.com/43583994/120485233-2eb3d880-c3b4-11eb-8773-b29db50d161c.png)
A symbol shows the different possibilities of construction for creatures' dwellings in the castle.
(This is similar to the system for different possibilities of construction in the castles: https://github.com/ihhub/fheroes2/discussions/3347.)

- Green color circle: you can build the creature's dwelling, all pre-required buildings are built, and you have the resources.
- Yellow color circle: you can build the creature's dwelling, all pre-required buildings are built, but you need to trade to have the resources.
- Orange color circle: you can build a pre-required building, you have the resources.
- Red color circle: you can build a pre-required building, but you need to trade to have the resources.
- Red color cross: you cannot build the creature's dwelling or a pre-required building, you don't have the resources and you cannot trade for them.

Picture 001.
![Kingdom info build dwelling H2 03a](https://user-images.githubusercontent.com/43583994/120485233-2eb3d880-c3b4-11eb-8773-b29db50d161c.png)
The Magi dwelling has already been built.

The green circle to the right of the growth number (+4) indicates that the upgraded dwelling can be built.

By left-clicking in the area indicated by the red rectangle, the building construction window is displayed (in the same way as the construction of the basic creature dwelling).
By left-clicking in the area below the area indicated by the red rectangle, it is possible to recruit creatures (as usual).

Picture 002.
![Overview build 02](https://user-images.githubusercontent.com/43583994/120120266-60eaed80-c19c-11eb-84b2-a1481fab58aa.png)
The building construction window is displayed.
For each pre-required building, the same symbols system is used to indicate the construction possibilities.
In the example, the Mage Guild Level 1 can be built directly (Green color circle), and the Foundry can be built by trading resources (Yellow color circle).

By left-clicking on the button with the green circle, the Mage Guild Level 1 construction window is displayed so the player can build (or not) the building.
By left-clicking on the button with the yellow circle, the castle screen is opened so the player can enter the Marketplace (in this castle, or in another castle with the arrows at the bottom of the screen) to trade resources and then build the Foundry.

This is an option:
"Game: display symbols according to the different possibilities of construction for creatures' dwellings"

**Comment by Kromster80** [comment]
I dislike both features, to be honest :-(
They make the game feel more like an Excel, than a piece of art it is.

**Reply by LeHerosInconnu** [reply]
Hello @Kromster80,

That's why it's offered as an option. :)
It's a quality of life improvement.
It saves a lot of back and forth between the screens because it allow to see at a glance the different possibilities.
